traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go

```diff
@@ -290,6 +290,57 @@ func GetDSNameAndCDNFromID(tx *sql.Tx, id int) (tc.DeliveryServiceName, tc.CDNNa
 	return dsName, cdnName, true, nil
 }
 
+// GetFederationResolversByFederationID fetches all of the federation resolvers currently assigned to a federation.
+func GetFederationResolversByFederationID(tx *sql.Tx, fedID int) ([]tc.FederationResolver, bool, error) {
+	qry := `
+		SELECT
+		  fr.ip_address,
+		  frt.name as resolver_type,
+		  ffr.federation_resolver
+		FROM
+		  federation_federation_resolver ffr
+		  JOIN federation_resolver fr ON ffr.federation_resolver = fr.id
+		  JOIN type frt on fr.type = frt.id
+		WHERE
+		  ffr.federation = $1
+	`
+	rows, err := tx.Query(qry, fedID)
+	if err != nil {
+		return nil, false, fmt.Errorf(
+			"error querying federation_resolvers by federation ID [%d]: %s", fedID, err.Error(),
+		)
+	}
+	defer rows.Close()
+
+	resolvers := []tc.FederationResolver{}
+	for rows.Next() {
+		fr := tc.FederationResolver{}
+		err := rows.Scan(
+			&fr.IPAddress,
+			&fr.Type,
+			&fr.ID,
+		)
+		if err != nil {
+			return nil, false, fmt.Errorf(
+				"error scanning federation_resolvers rows for federation ID [%d]: %s", fedID, err.Error(),
+			)
+		}
+		resolvers = append(resolvers, fr)
+	}
+	return resolvers, true, nil
+}
+
+// GetFederationNameFromID returns the federation's name.
+func GetFederationNameFromID(id int, tx *sql.Tx) (string, bool, error) {
+	var name string
+	if err := tx.QueryRow(`SELECT cname from federation where id = $1`, id).Scan(&name); err != nil {
+		return name, false, fmt.Errorf(
+			"error querying federation name from id [%d]: %s", id, err.Error(),
```

Comment:
You should check here for the error's equality with `database/sql.ErrNoRows` and in that event return `"", false, nil` to indicate that no actual error occurred, just that the requested object didn't exist. That allows the caller to act more intelligently without having to parse the error or check this equality itself - which is the ultimate point of that boolean return value.